csclient/csclient_test.go

```diff
@@ -296,35 +307,63 @@ func (s *suite) TestGetArchiveWithStatsDisabled(c *gc.C) {
 	key := s.checkGetArchive(c)
 
 	// Check that the downloads count for the entity has not been updated.
-	stats.CheckCounterSum(c, s.store, key, false, 0)
+	s.checkCharmDownloads(c, key, 0)
+}
+
+var checkDownloadsAttempt = utils.AttemptStrategy{
+	Total: 1 * time.Second,
+	Delay: 100 * time.Millisecond,
+}
+
+func (s *suite) checkCharmDownloads(c *gc.C, key string, expect int) {
+	stableCount := 0
+	for a := checkDownloadsAttempt.Start(); a.Next(); {
+		var result []params.Statistic
+		err := s.client.Get("/stats/counter/"+key, &result)
```

Comment:
That wasn't the plan (statsCount was just a relic) but actually it looks nicer using that, so done.